pkg/gui/controllers/helpers/commits_helper.go

```diff
@@ -0,0 +1,106 @@
+package helpers
+
+import (
+	"strings"
+
+	"github.com/jesseduffield/lazygit/pkg/gui/context"
+	"github.com/jesseduffield/lazygit/pkg/gui/types"
+)
+
+type ICommitsHelper interface {
+	SetMessageAndUpdateLength(message string, description string)
+}
+
+type CommitsHelper struct {
+	c *types.HelperCommon
+
+	model                *types.Model
+	contexts             *context.ContextTree
+	getCommitMessage     func() string
+	getCommitDescription func() string
+	refreshSuggestionsFn func()
+	setCommitMessage     func(string)
+	setCommitDescription func(string)
+	renderCommitLength   func()
+}
+
+var _ ICommitsHelper = &CommitsHelper{}
+
+func NewCommitsHelper(
+	c *types.HelperCommon,
+	model *types.Model,
+	contexts *context.ContextTree,
+	getCommitMessage func() string,
+	getCommitDescription func() string,
+	setCommitMessage func(string),
+	setCommitDescription func(string),
+	renderCommitLength func(),
+) *CommitsHelper {
+	return &CommitsHelper{
+		c:                    c,
+		model:                model,
+		contexts:             contexts,
+		getCommitMessage:     getCommitMessage,
+		getCommitDescription: getCommitDescription,
+		setCommitMessage:     setCommitMessage,
+		setCommitDescription: setCommitDescription,
+		renderCommitLength:   renderCommitLength,
+	}
+}
+
+func (self *CommitsHelper) SplitCommitMessageAndDescription(message string) (string, string) {
+	parts := strings.Split(message, "\n\n")
+	msg := parts[0]
+	var description string
+	if len(parts) > 1 {
+		description = strings.Join(parts[1:], "\n")
+	}
+	return msg, description
+}
+
+func (self *CommitsHelper) SetMessageAndUpdateLength(message string, description string) {
+	self.setCommitMessage(message)
+	self.setCommitDescription(description)
+	self.renderCommitLength()
+}
+
+func (self *CommitsHelper) GetCommitMessageAndDescription() (string, string) {
+	message := self.getCommitMessage()
+	description := self.getCommitDescription()
+	return message, description
+}
+
+func (self *CommitsHelper) getCommitPanelTitle(isRewording bool) string {
+	if isRewording == true {
```

Comment:
I want to see if we can do without this isRewording variable on the context.

Something that comes to mind:
* in the `PanelOptions` arg we could pass a title directly.
* we could pass a 'clearOnClose' boolean argument which could be stored on the context
* we can pass an onConfirm(message, description) argument which can also be stored on the context.

By making no reference to the reword action within the context itself we make our lives easier when some new use case comes along.

Also rather than call this `PanelOptions` I'd call it `OpenCommitMessagePanelOpts` to stick with convention